backend/routes.js

```diff
@@ -23,19 +34,75 @@ routes.getAllMessages = (req, res) => { // TYPE: GET
 };
 
 /*
-  //If the user logs in correctly, a JSON web token will be delivered. 
+//If the user logs in correctly, a JSON web token will be delivered.
 */
-routes.login = (req, res) => { // TYPE: "POST"
+routes.signin = (req, res) => { // TYPE: "POST"
+const username = req.body.username;
+const password = req.body.password;
+res.json({"response": "success"});
+}
+
+routes.signup = (req, res) => {
   const username = req.body.username;
   const password = req.body.password;
-  //Logic goes here
-  res.json({"response": "success"});
-}
+
+  //Checks to see if user didn't input username or password
+  if (typeof username === "undefined" || typeof password === "undefined") {
+    res.json({"response": "error", "errorType": "paramUndefined"})
+    return;
+  }
+
+  //If the user inputted nothing e.g. "   " or ""
+  if (username.trim() === "" || password.trim() === "") {
+    res.json({"response": "error", "errorType": "whitespace"});
+    return;
+  }
+
+  //If the username or pass is over 30 characters long
+  if (username.length > 30 || password.length > 30) {
+    res.json({"response": "error", "errorType": "tooLong"});
+    return;
+  }
+
+  //Creates a rando salt. Length of salt is 32 chars
+  const salt = crypto.randomBytes(16).toString("hex");
+  //Creates a sha1 hash of salt + password e.g. sha1(salt + password)
+  const hash = crypto.createHash('sha1').update(salt + password).digest('hex');
```

Comment:
This can be changed later when it matters, but as a general reminder SHA1 is not a secure method to store passwords.